### PR TITLE
ARROW-5507: [Plasma] [CUDA] Fix compile error

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

**Comment by pitrou** [comment]
AppVeyor build at https://ci.appveyor.com/project/pitrou/arrow/builds/25027414

File: cpp/src/plasma/protocol.cc [hunk 1]
```diff
@@ -543,7 +543,6 @@ Status SendGetReply(int sock, ObjectID object_ids[],
     if (object.device_num != 0) {
       std::shared_ptr<arrow::Buffer> handle;
       RETURN_NOT_OK(object.ipc_handle->Serialize(arrow::default_memory_pool(), &handle));
-      handles.push_back(fb::CreateCudaHandle(fbb, fbb.CreateVector(handle)));
```
Comment (emkornfield) [inline cpp/src/plasma/protocol.cc]: Consider using makenonnull to avoid ubsan issues

**Review by emkornfield — APPROVED** [review]
Sorry I thought I avoided cuda changes because I couldn't verify them.  Not near a computer so feel free to merge

**Comment by pitrou** [comment]
Rebasing.

**Comment by pitrou** [comment]
Since the one line changed here is in a part (CUDA-specific) that's not compiled on CI, I don't think it's necessary to wait for CI results.

**Comment by pitrou** [comment]
@emkornfield For the record, I'm not implying you should necessary buy a GPU, but there are cheap CUDA-enabled cards if you're interested. They are sufficient for testing. For example I have a passively-cooled GeForce GT 730.

**Pull request closed without merging**